### DISCUSSION TITLE
Is there a way to pass a class object to a new page?

### DISCUSSION THREAD

**Discussion by TommeZ** [lepoco/wpfui#278] [Q&A]
My page has a constructor that needs an object passed through to open. Is there a way to pass an object through using the page navigation in wpfui?

Class object = new Class();
Page myPage = new Page(object);
                (Application.Current.MainWindow as Container)?.RootNavigation.Navigate("myPage", object);

(The aforementioned code not taking the object I'm attempting to pass through)

**Comment by pomianowski** [comment]
**WPF UI** support **CommunityToolkit** and **Microsoft.Toolkit.Mvvm**. You can pass data to the page using dependency injection pattern.

You can see an example of use in the constructor here
https://github.com/lepoco/wpfui/blob/838cd35032e11d525585c8834ed79e01abf2d7ab/src/Wpf.Ui.Demo/Views/Pages/Dashboard.xaml.cs#L21

Dependency injection for WPF looks like this:
1. You register your pages and other repeatable objects in eg `App.cs` registering them as *services*.
2. In the constructor of the page (which is treated as a *service*) you declare to use a different services (e.g. `ViewModel` or `ThemeService`)
3. When navigation gets the `Page` object, it's constructor objects are passed via service provider.

In order for **WPF UI** to correctly understand pages as *services*, you need to create a special *service* named `IPageService`.
https://github.com/lepoco/wpfui/blob/838cd35032e11d525585c8834ed79e01abf2d7ab/src/Wpf.Ui.Demo/App.xaml.cs#L58

Then you need to tell your navigation control like `NavigationFluent` to use `IPageService`. Thanks to this, instead of creating new objects itself, it will use the service provider.
https://github.com/lepoco/wpfui/blob/838cd35032e11d525585c8834ed79e01abf2d7ab/src/Wpf.Ui.Demo/Views/Container.xaml.cs#L102
